### ISSUE TITLE
Nginx CrashLoopBackOff with 1.10.0

### ISSUE THREAD

**Issue by BlueCog** [RasaHQ/rasa-x-helm#191]
Hello,

I'm trying to deploy Rasa-X to our OpenShift environment. 

This OpenShift environment is closed down, so i had to import all the relevant images and replaced the pointing to this images by a helm override file.

The deployment seems to work allright with an exeption of the nginx pod:

I'm getting the following log output:

`/docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
/docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
10-listen-on-ipv6-by-default.sh: info: can not modify /etc/nginx/conf.d/default.conf (read-only file system?)
/docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
20-envsubst-on-templates.sh: ERROR: /etc/nginx/templates exists, but /etc/nginx/conf.d is not writable
/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
/docker-entrypoint.sh: Configuration complete; ready for start up
2021/05/04 10:14:26 [emerg] 1#1: mkdir() "/etc/nginx/client_body" failed (13: Permission denied)
nginx: [emerg] mkdir() "/etc/nginx/client_body" failed (13: Permission denied)
`

Note: In the past we were able to deploy Rasa-X without any problems. But this was with the custom rasa provided nginx image. 

has this something to do with the nature of OpenShift? Some online searching points me to changes in the nginx docker file (chown actions etc). But i'd like to prevent having to rebuild this nginx image myself...

Another solution is maybe to disable the nginx pod and using the routing functionality of OpenShift, but i'm not sure on how to implement this with Rasa-X.

Is there anyone who could put me in the right direction here?

**Comment by BlueCog** [comment]
Note:

Following the tips described here: https://number1.co.za/openshift-will-not-run-your-container-as-a-root-user/

I was able to deploy the nginx pod succesfully. 

So ar rebuild with a custom Docker file was needed:

```
RUN chgrp -R 0 /etc/nginx && \
    chmod -R g=u /etc/nginx
```

For me this seems like a workaround?

**Comment by sara-tagger** [comment]
Thanks for raising this issue, @desmarchris will get back to you about it soon✨

###### Please also check out the [docs](https://rasa.com/docs/) and the [forum](https://forum.rasa.com/) in case your issue was raised there too 🤗

**Comment by desmarchris** [comment]
@BlueCog this seems fine as a workaround, what user are you running everything else as?

**Comment by BlueCog** [comment]
> @BlueCog this seems fine as a workaround, what user are you running everything else as?

Hey @desmarchris thanks for getting back to me.

I'm deploying with an empty `securityContext`

This results in: `runAsUser: 1002880000 fsGroup: 1002880000 seLinuxOptions: s0:c54,c9`

**Comment by desmarchris** [comment]
hey @BlueCog I actually ran into this issue this week on OpenShift.

I was able to run the standard nginx image by changing my SCC to `anyuid` 

Regarding the other option: 
> Another solution is maybe to disable the nginx pod and using the routing functionality of OpenShift, but i'm not sure on how to implement this with Rasa-X.

if you set nginx install to false and ingress to true, this will take advantage of haproxy in openshift. the trick here is to make sure the domain name you provide for Rasa X matches the openshift nomenclature. For me this was: `<any string>.apps.<cluster-name>.<base-domain>`
